src/directives/repeat.js

```diff
@@ -683,6 +683,9 @@ module.exports = {
 
 function findPrevVm (vm, anchor, id) {
   var el = vm.$el.previousSibling
+  if (!el) {
```

Comment:
I finally got time to investigate this and it seems to indeed work! One thing here is that we can just `if (!el) return` here, no need for the empty object.